src/contracts/middleware/StakeRegistry.sol

```diff
@@ -0,0 +1,461 @@
+    }
+
+    /** 
+     * @notice Updates the stake for the operator with `operatorId` for the specified `quorumNumbers`. The total stake
+     * for each quorum is updated accordingly in addition to the operator's individual stake history.
+     */ 
+    function _registerOperator(address operator, bytes32 operatorId, bytes memory quorumNumbers) internal {
+        // check the operator is registering for only valid quorums
+        require(uint8(quorumNumbers[quorumNumbers.length - 1]) < quorumCount, "StakeRegistry._registerOperator: greatest quorumNumber must be less than quorumCount");
+        OperatorStakeUpdate memory _newTotalStakeUpdate;
+        // add the `updateBlockNumber` info
+        _newTotalStakeUpdate.updateBlockNumber = uint32(block.number);
+        // for each quorum, evaluate stake and add to total stake
+        for (uint8 quorumNumbersIndex = 0; quorumNumbersIndex < quorumNumbers.length;) {
+            // get the next quorumNumber
+            uint8 quorumNumber = uint8(quorumNumbers[quorumNumbersIndex]);
+            // evaluate the stake for the operator
+            // since we don't use the first output, this will use 1 extra sload when deregistered operator's register again
+            (, uint96 stake) = _updateOperatorStake(operator, operatorId, quorumNumber);
+            // @JEFF: This reverts pretty late, but i think that's fine. wdyt?
+            // check if minimum requirement has been met, will be 0 if not
+            require(stake != 0, "StakeRegistry._registerOperator: Operator does not meet minimum stake requirement for quorum");
+            // add operator stakes to total stake before update (in memory)
+            uint256 _totalStakeHistoryLength = _totalStakeHistory[quorumNumber].length;
+            // add calculate the total stake for the quorum
+            uint96 totalStakeAfterUpdate = stake;
+            if (_totalStakeHistoryLength != 0) {
+                // only add the stake if there is a previous total stake
+                // overwrite `stake` variable
+                totalStakeAfterUpdate += _totalStakeHistory[quorumNumber][_totalStakeHistoryLength - 1].stake;
+            }
+            _newTotalStakeUpdate.stake = totalStakeAfterUpdate;
+            // update storage of total stake
+            _recordTotalStakeUpdate(quorumNumber, _newTotalStakeUpdate);
+            unchecked {
+                ++quorumNumbersIndex;
+            }
+        }
+    }
+
+    /**
+     * @notice Removes the stakes of the operator with `operatorId` from the quorums specified in `quorumNumbers`
+     * the total stake of the quorums specified in `quorumNumbers` will be updated and so will the operator's individual
+     * stake updates. These operator's individual stake updates will have a 0 stake value for the latest update.
+     */
+    function _deregisterOperator(bytes32 operatorId, bytes memory quorumNumbers) internal {
+        // check the operator is deregistering from only valid quorums
```

Comment:
the comment here is erroneous / no such check is occurring -- I'm assuming the check has moved higher up the call stack?